packages/lesswrong/lib/collections/posts/tableOfContents.js

```diff
@@ -56,14 +56,17 @@ export function extractTableOfContents(postHTML, alwaysDisplayToC)
     }
 
     let title = cheerio(tag).text();
-    let anchor = titleToAnchor(title, usedAnchors);
-    usedAnchors[anchor] = true;
-    cheerio(tag).attr("id", anchor);
-    headings.push({
-      title: title,
-      anchor: anchor,
-      level: tagToHeadingLevel(tag.tagName),
-    });
+    
+    if (title && title.trim()!=="") {
```

Comment:
I think I would prefer a length check over a compare with empty string, but I guess either is fine